noir-projects/aztec-nr/aztec/src/state_vars/single_use_claim/test.nr

```diff
@@ -0,0 +1,173 @@
+use crate::{
+    context::{private_context::PrivateContext, UtilityContext},
+    oracle::random::random,
+    state_vars::single_use_claim::SingleUseClaim,
+};
+use crate::test::{helpers::test_environment::TestEnvironment, mocks::mock_struct::MockStruct};
+use dep::protocol_types::{address::AztecAddress, traits::FromField};
+
+global STORAGE_SLOT: Field = 17;
+
+unconstrained fn in_private(
+    context: &mut PrivateContext,
+) -> SingleUseClaim<MockStruct, &mut PrivateContext> {
+    SingleUseClaim::new(context, STORAGE_SLOT)
+}
+
+unconstrained fn in_utility(
+    context: UtilityContext,
+    storage_slot: Field,
+) -> SingleUseClaim<MockStruct, UtilityContext> {
+    SingleUseClaim::new(context, storage_slot)
+}
+
+unconstrained fn expect_claim_to_exist(
+    claim_exists: bool,
+    env: TestEnvironment,
+    storage_slot: Field,
+    claim_id: MockStruct,
+    owner: AztecAddress,
+) {
+    env.utility_context(|context| {
+        let state_var = in_utility(context, storage_slot);
+
+        assert(state_var.at(owner).has_claimed(claim_id) == claim_exists);
+    });
+}
+
+unconstrained fn expect_not_yet_claimed(
+    env: TestEnvironment,
+    storage_slot: Field,
+    claim_id: MockStruct,
+    owner: AztecAddress,
+) {
+    expect_claim_to_exist(false, env, storage_slot, claim_id, owner)
+}
+
+unconstrained fn expect_claimed(
+    env: TestEnvironment,
+    storage_slot: Field,
+    claim_id: MockStruct,
+    owner: AztecAddress,
+) {
+    expect_claim_to_exist(true, env, storage_slot, claim_id, owner)
+}
+
+#[test(should_fail_with = "Duplicate siloed nullifier")]
+unconstrained fn claim_twice_same_tx() {
+    let mut env = TestEnvironment::new();
+
+    let owner = env.create_light_account();
+    let claim_id = MockStruct::new(1, 2);
+
+    env.private_context(|context| {
+        let state_var = in_private(context);
+
+        state_var.at(owner).claim(claim_id);
+        state_var.at(owner).claim(claim_id);
+    });
+}
+
+#[test]
+unconstrained fn has_not_yet_claimed() {
+    let mut env = TestEnvironment::new();
+    let owner = env.create_light_account();
+
+    let claim_id = MockStruct::new(1, 2);
+    expect_not_yet_claimed(env, STORAGE_SLOT, claim_id, owner);
```

Comment:
This helper is so simple that I'm not 100% sold on it being better than just inlining the contents - I ended up going back and forth to the definition to remind myself what it is the helper did exactly. Up to you though.